### PR TITLE
Split out functions in unknown_070110, and do some documenting on the…

### PR DESCRIPTION
**Pull request by Ryan-Myers**
Split out functions in unknown_070110, and do some documenting on them. Also fixes the scoring script to include UNUSED funcs.

### CODE REVIEW

File: asm/unknown_070110/func_800708D0.s [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+/* 0715E4 800709E4 A099FFFE */  sb    $t9, -2($a0)
+/* 0715E8 800709E8 5580FFD6 */  bnezl $t4, .L80070944
+/* 0715EC 800709EC 00000000 */   nop   
+/* 0715F0 800709F0 2508FFFF */  addiu $t0, $t0, -1
+/* 0715F4 800709F4 5500FFBE */  bnezl $t0, .L800708F0
+/* 0715F8 800709F8 00000000 */   nop   
+/* 0715FC 800709FC 03E00008 */  jr    $ra
+/* 071600 80070A00 00000000 */   nop   
+
+/* 071604 80070A04 94CC0002 */  lhu   $t4, 2($a2)
```
Comment (AntonioCastelli) [inline asm/unknown_070110/func_800708D0.s:84]: Make this a standalone function?

File: include/structs.h [hunk 2]
```diff
@@ -1174,6 +1174,16 @@ typedef struct Object_8001B7A8 {
   /* 0x112 */ s16 unk112;
 } Object_8001B7A8;
 
+typedef struct Object_64_80021400 {
+  /* 0x00 */ u8 pad[0x29];
```
Comment (AntonioCastelli) [inline include/structs.h:1178]:
```suggestion
  /* 0x00 */ u8 pad0[0x2A];
```

File: include/structs.h [hunk 3]
```diff
@@ -1174,6 +1174,16 @@ typedef struct Object_8001B7A8 {
...
+  /* 0x2A */ s16 unk2A;
+} Object_64_80021400;
+
+typedef struct Object_80021400 {
+  /* 0x000 */ u8 pad[0x63];
```
Comment (AntonioCastelli) [inline include/structs.h:1183]:
```suggestion
  /* 0x000 */ u8 pad0[0x64];
```

File: include/structs.h [hunk 4]
```diff
@@ -1217,6 +1227,7 @@ typedef struct Object_64 {
         Object_Bridge_WhaleRamp bridge_whale_ramp;
         Object_80011AD0 obj80011AD0;
         Object_8001B7A8 obj8001B7A8;
+        Object_80021400 obj80021400;
```
Comment (AntonioCastelli) [inline include/structs.h:1230]: These are supposed to be a union of all the unk64 structs. Should this instead be a `Object_64_80021400`? Or does the unk64 struct itself contain a struct pointer at offset 0x64?

File: src/objects.c [hunk 5]
```diff
@@ -1548,7 +1549,22 @@ s32 func_800210CC(s8 arg0) {
 
 GLOBAL_ASM("asm/non_matchings/objects/func_80021104.s")
 GLOBAL_ASM("asm/non_matchings/objects/func_8002125C.s")
-GLOBAL_ASM("asm/non_matchings/objects/func_80021400.s")
+
+void func_80021400(s32 arg0) {
+    s32 i;
+    arg0 &= 0xFF; //?
+
```
Comment (AntonioCastelli) [inline src/objects.c:1556]:
```suggestion
    arg0 &= 0xFF;
```

I also assume that the function doesn't match if it's a u8/s8?

**Review by AntonioCastelli — APPROVED** [review]

File: asm/unknown_070110/func_800708D0.s (hunk 1 above)
Comment (Ryan-Myers) [reply asm/unknown_070110/func_800708D0.s:84]: Hmm, it's a bit of an odd one. Lends credence to it's most likely state as a hand written chunk of assembly though. On line 85 of that file it has the label: `L80070A08` which is jumped to from the line 13 in `func_800708D0`. But yeah, that line 84 lhu suggests that there's a function there that is just unused that would start on that instruction and then lead into the next ones on it's own, and then oddly jump back into the middle of `func_800708D0` from line 91. So this is just a weird blend of functions that can jump in and out of each other in a way that only hand written assembly could do?

File: src/objects.c (hunk 5 above)
Comment (Ryan-Myers) [reply src/objects.c:1556]: Yeah, it's weird. If I change it to a u8 / s8 then the diff looks like this:
image

File: asm/unknown_070110/func_800708D0.s (hunk 1 above)
Comment (AntonioCastelli) [reply asm/unknown_070110/func_800708D0.s:88]: Ah, missed the jump into this from above. Leaving together sgtm, thanks.

File: include/structs.h (hunk 4 above)
Comment (AntonioCastelli) [reply include/structs.h:1232]: Gotcha. In that case can you rename `Object_64_80021400` to `Object_80021400_64`? That way it's not confused as an Object64, and the left-to-right nature of reading implies the structure of the type.

**Review by AntonioCastelli — APPROVED** [review]

File: include/sound_ids.h [hunk 6]
```diff
@@ -12,648 +12,648 @@
...
+/* 0x00F7 */ SOUND_VOICE_TRICKY_LAUGH2,
+/* 0x00F8 */ SOUND_VOICE_SMOKEY_HAH,
+/* 0x00F9 */ SOUND_VOICE_SMOKEY_LAUGH,
+/* 0x00FA */ SOUND_VOICE_BLUEY_HAHA3,
+/* 0x00FB */ SOUND_VOICE_BLUEY_HAHA4,
+/* 0x00FC */ SOUND_UNK_FC,
+/* 0x00FD */ SOUND_CROWD,
+/* 0x00FE */ SOUND_MENU_PICK4,
+/* 0x00FF */ SOUND_UNK_FF,
+/* 0x0100 */  SOUND_VOICE_TT_LAP2,
```
Comment (AntonioCastelli) [inline include/sound_ids.h:271]: Looks like there's some indentation weirdness here and below.

**Review by AntonioCastelli — APPROVED** [review]

File: include/sound_ids.h (hunk 6 above)
Comment (Ryan-Myers) [reply include/sound_ids.h:271]: Fixed.

File: include/sound_ids.h [hunk 7]
```diff
@@ -12,648 +12,648 @@
...
+/* 0x0217 */  SOUND_VOICE_BUMPER_NEGATIVE8,
+/* 0x0218 */  SOUND_VOICE_TIPTUP_NEGATIVE8,
+/* 0x0219 */  SOUND_VOICE_CONKER_NEGATIVE8,
+/* 0x021A */  SOUND_VOICE_TIMBER_NEGATIVE8,
+/* 0x021B */  SOUND_VOICE_BANJO_NEGATIVE8,
+/* 0x021C */  SOUND_VOICE_DRUMSTICK_NEGATIVE8, // Sounds kind of profane ngl
+/* 0x021D */  SOUND_VOICE_PIPSY_NEGATIVE8,
+/* 0x021E */  SOUND_VOICE_TT_NEGATIVE8,
+/* 0x021F */  SOUND_VOICE_DIDDY_NEGATIVE8,
+             //Betting these are unused, because they sit in the spot where character ID's should be.
```
[the remark this answers is not in the document]
Comment (Ryan-Myers) [reply include/sound_ids.h:559]: Yeah, I'm not sure what that comment was for (not written by me). I'll just remove it.

File: include/structs.h [hunk 8]
```diff
@@ -6,12 +6,23 @@
...
-    f32 y;
-    f32 z;
+	union {
+		struct {
+			f32 x;
+			f32 y;
+			f32 z;
+		};
+		f32 f[3];
+	};
```
[the remark this answers is not in the document]
Comment (Ryan-Myers) [reply include/structs.h:22]: Woops, probably an issue with tabs vs spaces from the PD repo I stole it from. Fixed.

**Pull request merged**